src/SQLite.Net/TableQuery.cs

```diff
@@ -484,6 +505,28 @@ private CompileResult CompileExpr( Expression expr, List<object> queryArgs)
                         CommandText = "\"" + columnName + "\""
                     };
                 }
+                
+                if (mem.Expression != null && ExpressionIsTupleAccess(mem))
+                {
+                    // This only supports a single level of nesting. That is
+                    // x => x.Key.item is allowed
+                    // but x => x.Key.item.value is not allowed.
+                    
+                    var parentProperty = (MemberExpression)mem.Expression;
+                    var memberName = mem.Member.Name;
+                    var names = parentProperty.Member.GetCustomAttribute<TupleElementNamesAttribute>();
+                    if (names != null)
+                    {
+                        var index = int.Parse(memberName.Substring(4)) - 1;
+                        memberName = names.TransformNames[index];
+                    }
+                    var name = parentProperty.Member.Name + "_" + memberName;
+                    return new CompileResult
+                    {
+                        CommandText = "\"" + name + "\""
+                    };
+                }
```

Comment:
can you explain this block im confused

Reply:
Added some more comments and renamed the method. Hopefully this is more clear?